core/src/main/scala/com/salesforce/op/filters/FeatureDistribution.scala

```diff
@@ -99,10 +101,25 @@ case class FeatureDistribution
    */
   def reduce(fd: FeatureDistribution): FeatureDistribution = {
     checkMatch(fd)
+    // should move this somewhere else
+    implicit val testStatsSG: Semigroup[TextStats] = TextStats.semiGroup(FeatureDistribution.MaxCardinality)
     val combinedDist = distribution + fd.distribution
     // summary info can be empty or min max if hist is empty but should otherwise match so take the longest info
-    val combinedSummary = if (summaryInfo.length > fd.summaryInfo.length) summaryInfo else fd.summaryInfo
-    FeatureDistribution(name, key, count + fd.count, nulls + fd.nulls, combinedDist, combinedSummary, `type`)
+    val combinedSummaryInfo = if (summaryInfo.length > fd.summaryInfo.length) summaryInfo else fd.summaryInfo
+    val combinedMoments = (moments, fd.moments) match {
+      case (Some(x), None) => Some(x)
+      case (Some(x), Some(y)) => Some(x + y)
+      case (None, Some(y)) => Some(y)
+      case (_, _) => None
+    }
```

Comment:
do you really need to write all this out when you have operators imported? I think this should be the default behavior with operators.

Reply:
I got this error `diverging implicit expansion for type algebra.Monoid[com.salesforce.op.stages.impl.feature.TextStats]
starting with method fromAlgebraMonoid in trait FromAlgebraMonoidImplicit0
` if I simply do `moments + fd.moments` or `value + is not a member of Option[com.salesforce.op.stages.impl.feature.TextStats]` for`cardEstimate + fd.cardEstimate` (`testStatsSG.plus` only works on non-Option `TextStats` objects)

Reply:
maybe try defining the implicit function like here: https://github.com/salesforce/TransmogrifAI/blob/master/core/src/main/scala/com/salesforce/op/filters/RawFeatureFilter.scala#L156

Reply:
```
implicit val sgTuple2Moments = new Tuple2Semigroup[Moments, Moments]()
implicit val sgTuple2TextStats = new Tuple2Semigroup[TextStats, TextStats]()
```
i tried doing this ^ and got a bunch of errors 

Error:(110, 38) diverging implicit expansion for type com.twitter.algebird.Semigroup[com.salesforce.op.stages.impl.feature.TextStats]
starting with method fromAlgebraMonoid in trait FromAlgebraMonoidImplicit0
    implicit val sgTuple2TextStats = new Tuple2Semigroup[TextStats, TextStats]()
Error:(110, 38) not enough arguments for constructor Tuple2Semigroup: (implicit asemigroup: com.twitter.algebird.Semigroup[com.salesforce.op.stages.impl.feature.TextStats], implicit bsemigroup: com.twitter.algebird.Semigroup[com.salesforce.op.stages.impl.feature.TextStats])com.twitter.algebird.Tuple2Semigroup[com.salesforce.op.stages.impl.feature.TextStats,com.salesforce.op.stages.impl.feature.TextStats].
Unspecified value parameters asemigroup, bsemigroup.
    implicit val sgTuple2TextStats = new Tuple2Semigroup[TextStats, TextStats]()
Error:(112, 50) diverging implicit expansion for type algebra.Monoid[com.salesforce.op.stages.impl.feature.TextStats]
starting with method fromAlgebraMonoid in trait FromAlgebraMonoidImplicit0
      combinedSummaryInfo, moments + fd.moments, cardEstimate + fd.cardEstimate, `type`)
Error:(112, 50) diverging implicit expansion for type algebra.ring.AdditiveMonoid[com.salesforce.op.stages.impl.feature.TextStats]
starting with method fromAlgebraMonoid in trait FromAlgebraMonoidImplicit0
      combinedSummaryInfo, moments + fd.moments, cardEstimate + fd.cardEstimate, `type`)
Error:(112, 50) diverging implicit expansion for type algebra.Monoid[Option[com.salesforce.op.stages.impl.feature.TextStats]]
starting with method fromAlgebraMonoid in trait FromAlgebraMonoidImplicit0
      combinedSummaryInfo, moments + fd.moments, cardEstimate + fd.cardEstimate, `type`)
Error:(112, 63) value + is not a member of Option[com.salesforce.op.stages.impl.feature.TextStats]
      combinedSummaryInfo, moments + fd.moments, cardEstimate + fd.cardEstimate, `type`)

Reply:
yes this is not a tuple - it is an option

Reply:
@tovbinm any idea how to get the implicits to work so this doesnt have to be written out?

Reply:
@leahmcguire adding `implicit val opMonoid = optionMonoid[TextStats]` let me do `val combinedMoments = moments + fd.moments` and the tests passed

Reply:
Ooo, I think I fixed it!

I was digging into Algebirds monoid construction code and it looked like it was expecting an implicit `Monoid[TextStats]`, even though the helper function only asked for a SemiGroup.

I added a monoid to out `TextStats` and then it worked fine with
```
implicit val testStatsMonoid: Monoid[TextStats] = TextStats.monoid(RawFeatureFilter.MaxCardinality)
implicit val opMonoid = optionMonoid[TextStats]
```

I don't actually know why we made `TextStats` a SemiGroup instead of a monoid in the first place, the only difference between the two is an identity element and we already explicitly defined a zero element for `TextStats`.